lib/dependencies/RequireEnsureDependency.js

```diff
@@ -2,24 +2,30 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
-var DepBlockHelpers = require("./DepBlockHelpers");
+"use strict";
+const NullDependency = require("./NullDependency");
+const DepBlockHelpers = require("./DepBlockHelpers");
 
-function RequireEnsureDependency(block) {
-	NullDependency.call(this);
-	this.block = block;
-}
-module.exports = RequireEnsureDependency;
+class RequireEnsureDependency extends NullDependency {
+	constructor(block) {
+		super();
+		this.block = block;
+	}
 
-RequireEnsureDependency.prototype = Object.create(NullDependency.prototype);
-RequireEnsureDependency.prototype.constructor = RequireEnsureDependency;
-RequireEnsureDependency.prototype.type = "require.ensure";
+	get type() {
+		return "require.ensure";
```

Comment:
@timse that puts it in the constructor, though; should be `RequireEnsureDependency.prototype.type = "require.ensure";`

@chicoxyzzy I kinda prefer the approach of always setting in the constructor if only because that's how https://tc39.github.io/proposal-class-public-fields/ desugars, but if it's a constant that's never changed per instance it might as well be put specifically in the prototype.